lib/pbench/server/api/resources/upload_api.py

```diff
@@ -135,6 +146,34 @@ def _put(self, args: ApiParams, request: Request, context: ApiContext) -> Respon
         access = (
             args.query["access"] if "access" in args.query else Dataset.PRIVATE_ACCESS
         )
+
+        # We allow the client to set metadata on the new dataset. We won't do
+        # anything about this until upload is successful, but we process and
+        # validate it here so we can fail early.
+        metadata = {}
+        if "metadata" in args.query:
+            errors = []
+            for kw in args.query["metadata"]:
```

Comment:
if there are multiple values in `metadata` like `"foobar.badpath:data,server.deletion:3000-12-25T23:59:59+00:00"` will it take `foobar.badpath:data` and `server.deletion:3000-12-25T23:59:59+00:00` separately?

Reply:
Yes. Of course it'll fail because `foobar.badpath` isn't a valid metadata key. 😁 Also, this currently only handles *string* values; you can't set integer values, or boolean values, or JSON objects. I debated internally how/whether to handle that, and just didn't.